src/Stratis.Bitcoin.Features.BlockStore/AddressIndexing/AddressIndexer.cs

```diff
@@ -510,75 +500,60 @@ public bool IsSynced()
         }
 
         /// <inheritdoc />
-        public Money GetAddressBalance(string address, int minConfirmations = 1)
+        public AddressBalancesResult GetAddressBalances(string[] addresses, int minConfirmations = 1)
         {
-            if (this.addressIndexRepository == null)
-            {
-                this.logger.LogTrace("(-)[NOT_INITIALIZED]");
-                throw new IndexerNotInitializedException();
-            }
+            var (isQuerable, reason) = this.IsQuerable();
+            if (!isQuerable)
+                return AddressBalancesResult.RequestFailed(reason);
 
-            if (!this.IsSynced())
-            {
-                this.logger.LogTrace("(-)[NOT_SYNCED]");
-                throw new OutOfSyncException();
-            }
+            var result = new AddressBalancesResult();
 
             lock (this.lockObject)
             {
-                AddressIndexerData indexData = this.addressIndexRepository.GetOrCreateAddress(address);
-                if (indexData == null)
+                foreach (var address in addresses)
                 {
-                    this.logger.LogTrace("(-)[NOT_FOUND]");
-                    return null;
-                }
+                    AddressIndexerData indexData = this.addressIndexRepository.GetOrCreateAddress(address);
```

Comment:
Note that, as the name suggests, this will return an `AddressIndexerData` with an empty list of balance changes if it is not in the cache. It shouldn't ever return `null`.

Reply:
Ah OK I will remove this 👍